source/server/admin/grouped_stats_request.h

```diff
@@ -0,0 +1,56 @@
+#pragma once
+
+#include <vector>
+
+#include "envoy/server/admin.h"
+
+#include "source/server/admin/stats_params.h"
+#include "source/server/admin/stats_render.h"
+#include "source/server/admin/stats_request.h"
+#include "source/server/admin/utils.h"
+
+#include "absl/container/btree_map.h"
+#include "absl/types/variant.h"
+
+namespace Envoy {
+namespace Server {
+
+class GroupedStatsRequest
```

Comment:
a class comment would be good, something like:

// Renders stats grouped by tag-extracted name. This is currently used for Prometheus (and has some
// Prometheus-specific logic in it) but this could be generalized for other formats.